src/repositories/users.ts

```diff
@@ -108,3 +108,25 @@ export async function changeRole(userId: number, roleType: number) {
       WHERE user_id = ${userId}`);
   });
 }
+
+export async function deleteById(userId: number): Promise<void> {
+  return DatabaseConnection.getConnectionPool().connect(async connection => {
+    await connection.transaction(async transaction => {
+      Promise.all([
```

Comment:
надо повесить await на этот Promise.all, иначе пользователи могут удалиться раньше, чем ассоциации, что вызовет ошибку. 
```suggestion
      await Promise.all([
```